### PR TITLE
Issues #629, #681 (partly) regarding check_ncpa.py

### PR DESCRIPTION
**Pull request by ccztux**
This pull request implements:
- [x] Add perfdata prefix/suffix (#629)
- [x] Output conversion from multiline output to a single line of html (https://github.com/NagiosEnterprises/ncpa/issues/681#issuecomment-687630704)

### CODE REVIEW

File: client/check_ncpa.py [hunk 1]
```diff
@@ -318,10 +404,6 @@ def main():
     signal.alarm(options.timeout)
 
     try:
-        if options.version:
-            stdout = 'The version of this plugin is %s' % __VERSION__
-            return stdout, 0
-
```
Comment (ccztux) [inline client/check_ncpa.py:324]: Removed, will already be checked in function parse_args.

File: client/check_ncpa.py [hunk 2]
```diff
@@ -53,18 +53,19 @@
...
-__VERSION__ = '1.2.2'
+__VERSION__ = '1.3.0'
 
 
 class ConnectionError(Exception):
     error_output_prefix = "UNKNOWN: An error occured connecting to API. "
     pass
 
+
```
Comment (ccztux) [inline client/check_ncpa.py:63]: PEP8 (E302)

File: client/check_ncpa.py [hunk 3]
```diff
@@ -350,5 +445,5 @@ def main():
 
 if __name__ == "__main__":
     stdout, returncode = main()
-    print(stdout.encode('utf-8', 'replace'))
+    print(stdout.encode('utf-8', 'replace').decode('utf-8'))
```
Comment (ccztux) [inline client/check_ncpa.py:448]: Merged from newer version 1.2.3 in master branch.

File: client/check_ncpa.py [hunk 4]
```diff
@@ -53,18 +53,19 @@
...
 class ConnectionError(Exception):
     error_output_prefix = "UNKNOWN: An error occured connecting to API. "
     pass
 
+
 class URLError(ConnectionError):
     def __init__(self, error_message):
         self.error_message = ConnectionError.error_output_prefix + "(Connection error: '" + error_message + "')"
 
+
```
Comment (ccztux) [inline client/check_ncpa.py:68]: PEP8 (E302)

File: client/check_ncpa.py [hunk 5]
```diff
@@ -73,61 +74,78 @@ def __init__(self, error_message):
...
+                        help='Add a prefix to perfdata labels.')
+    perfdata.add_option("-B", "--perfdata-suffix", action='store', type="string",
+                        help='Add a suffix to perfdata labels.')
+    parser.add_option_group(perfdata)
+
+    output = optparse.OptionGroup(parser, "Output Conversion Options")
+    output.add_option("-S", "--html-single-line", action='store_true', default=False,
+                      help='Convert multi line output to single line html output.')
+    parser.add_option_group(output)
+
```
Comment (ccztux) [inline client/check_ncpa.py:141]: Options grouped and PEP8 (E127)

File: client/check_ncpa.py [hunk 6]
```diff
@@ -73,61 +74,78 @@ def __init__(self, error_message):
...
     options, _ = parser.parse_args()
 
     if options.version:
         print(version)
         sys.exit(0)
 
-    if options.arguments and options.metric and not 'plugin' in options.metric:
+    if options.arguments and options.metric and 'plugin' not in options.metric:
```
Comment (ccztux) [inline client/check_ncpa.py:148]: PEP8 (E713)

File: client/check_ncpa.py [hunk 7]
```diff
@@ -162,7 +184,7 @@ def get_host_part_from_options(options):
     hostname = options.hostname
     port = options.port
 
-    if not options.metric is None:
+    if options.metric is not None:
```
Comment (ccztux) [inline client/check_ncpa.py:187]: PEP8 (E714)

File: client/check_ncpa.py [hunk 8]
```diff
@@ -201,9 +223,9 @@ def get_arguments_from_options(options, **kwargs):
 
     # Note: Changed back to units due to the units being what is passed via the
     # API call which can confuse people if they don't match
-    arguments = { 'token': options.token,
-                  'units': options.units }
-    
+    arguments = {'token': options.token,
+                 'units': options.units}
+
```
Comment (ccztux) [inline client/check_ncpa.py:228]: PEP8 (E201, E202)

**Comment by ccztux** [comment]
@jomann09 I have some questions regarding this pull request.
What do you think about adding perfdata pre- and suffixes to the output provided by the ```plugins``` node? Should this be allowed by ```check_ncpa.py```, or should it be denied?

As far as i know, all NCPA internal checks, except the ```plugins``` node provides perfdata always in the first line of output, also in case of multiline output, like in the following example (not from a NCPA check ;) ):
```
DISK OK - free space: / 3326 MB (56%); | /=2643MB;5948;5958;0;5968
/ 15272 MB (77%);
/boot 68 MB (69%);
/home 69357 MB (27%);
/var/log 819 MB (84%);
```

Can you confirm this?

**Comment by ccztux** [comment]
I have tested the following and in my opinion not supported by nagios output and perfdata construct with nagios 4.4.6:
```
DISK OK - free space: / 3326 MB (56%); | /=2643MB;5948;5958;0;5968
/ 15272 MB (77%); | blah=10MB;1;2;;; puh=1MB;1;2;;
/boot 68 MB (69%); | third_line=1;;;;
/home 69357 MB (27%);
/var/log 819 MB (84%); | /boot=68MB;88;93;0;98
/home=69357MB;253404;253409;0;253414
/var/log=818MB;970;975;0;980
```

And this is what nagios shows in the web UI:
Bildschirmfoto vom 2020-09-17 20-22-59

Can you confirm, that this is not supported by nagios as described in the [documentation](https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/4/en/pluginapi.html)

**Comment by jomann09** [comment]
We are following the plugin guidelines for Nagios when it comes to the output. Which is here https://nagios-plugins.org/doc/guidelines.html. The guidelines say that you can put perfdata on the last line as well as the first line, so I really don't see a problem with it. However you are correct that by default, all perfdata generated by NCPA itself and passed back to the plugin is passed via the first line only.

**Pull request closed without merging**